### ISSUE TITLE
Table name automatically capitalizes internal table name first letter

### ISSUE THREAD

**Issue by PeterCDMcLean** [gristlabs/grist-core#755]
Not a bug per-se, but surprising behaviour. 

Upon renaming a table:

![image](https://github.com/gristlabs/grist-core/assets/6133132/0f475722-94bf-45a1-ad28-090f0d1033e8)

The internal table name automatically capitalizes the first letter:

![image](https://github.com/gristlabs/grist-core/assets/6133132/d83ed696-58cd-4b11-81e9-36c7818239a5)

**Comment by paulfitz** [comment]
Yes, that's a little awkward. There are a few other changes that can happen also to the name you set, such as spaces becoming underscores. For columns, the distinction between a "column label" (which can be anything) and a "column  id" (which we try to make as similar as possible subject to some constraints) is made fairly clear in the UI, and can be explicitly controlled by the user. The raw data view makes a distinction between "table name" and "table id", but does not offer a way to control the id. I wonder if it would be practical to allow that - do you know @dsagal ?

**Comment by dsagal** [comment]
We could allow controlling the tableId, yes. The reasons for the capitalization are:

1. consistency -- having all tables capitalized makes it easier to tell that a particular identifier probably refers to a table
2. since table IDs are available as global variables in Python (for formulas like `People.lookupOne(...)` or `Departments.all.Name`), a lowercase name would open up conflicts with other Python global identifiers, like `property` or `dir`, and some names would not be allowed at all (like `class`).

So I think capitalizing the table ID is still a good default, but allowing to manually override it (taking care to avoid Python keywords) seems OK to me.
